src/lib.rs

```diff
@@ -68,84 +78,63 @@ impl<T> From<cc::SendError<T>> for RequestError {
     }
 }
 
-/// A object expected tois a request which is received from the [Responder] poll method
-///
-/// The request body can be obtained from the body() function and before being
-/// dropped it needs to send a response with the respond() function.
-/// Not doing a response on a request is considered a programmer error and will result in a panic
-/// when the object gets dropped
-pub struct Request<Req, Res> {
-    request: Req,
+/// A [ResponseSender] is received from the [RequestReceiver] to respond to the request back to the
+/// [RequestSender]
+pub struct ResponseSender<Res> {
     response_sender: cc::Sender<Res>,
-    _responded: bool
 }
 
-impl<Req, Res> Request<Req, Res> {
-    fn new(request: Req, response_sender: cc::Sender<Res>) -> Request<Req, Res> {
-        Request {
-            request: request,
+impl<Res> ResponseSender<Res> {
+    fn new(response_sender: cc::Sender<Res>) -> ResponseSender<Res> {
+        ResponseSender {
             response_sender: response_sender,
-            _responded: false,
         }
     }
 
-    /// Get actual request data
-    pub fn body(&self) -> &Req {
-        &self.request
-    }
-
-    /// TODO
-    pub fn respond(&mut self, response: Res) {
-        if self._responded {
-            panic!("Programmer error, same request cannot respond twice!");
-        }
+    /// Responds a request from the [RequestSender] which finishes the request
+    pub fn respond(&self, response: Res) {
         match self.response_sender.send(response) {
             Ok(_) => (),
-            Err(_e) => panic!("Request failed, send pipe was broken during request!")
-        }
-        self._responded = true;
-    }
-}
-
-impl<Req, Res> Drop for Request<Req, Res> {
-    fn drop(&mut self) {
-        if !self._responded {
-            panic!("Dropped request without responding, programmer error!");
+            Err(_e) => panic!("Response failed, send pipe was broken during request!")
```

Comment:
What happens if the responder responds twice?

Reply:
Nothing really, before it was just an artificial limitation. As long as the receiver isn't dropped while it's sending a message it will be fine which is the same as before.